src/function/FunctionGraph.ts

```diff
@@ -232,6 +232,29 @@ export function setRepresentation(representation: number): { result: boolean, tr
     }
 }
 
+export function checkLinkSelfLoop(link: joint.dia.Link) {
```

Comment:
Two minor things here, up to you if you modify the code:
1) The function probably does not need to be exported since it's only use seems to be in this file.
2) I'd consider a different name. It's weird to have something named "check..." and then return an array of points.